### PR TITLE
Extract ContentShell component

### LINKED ISSUE #730: Move `ContentShell` component to its own file
Supports #588

---

This is needed because we'll likely connect this component as well as the `FullscreenGrid` component. There will be too much code into this `FullscreenGrid.tsx` file, so let's split it.

**For QA:** there should be no visual regression, i.e. pages should show 3 panels (left panel usually has the accordion menu, center panel has code, right panel has a code overview).

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #730

---

This patch supports what's described in #721. I kept the `ContentShell`
files in `FullscreenGrid/` because it is strongly coupled.

### CODE REVIEW

File: src/components/FullscreenGrid/index.tsx [hunk 1]
```diff
@@ -2,8 +2,18 @@ import * as React from 'react';
...
 import styles from './styles.module.scss';
 
+type PublicProps = {
+  children?: AnyReactNode;
+  className?: string;
+};
+
+type Props = PublicProps;
+
+export type PanelAttribs = 'altSidePanel' | 'mainSidePanel';
```
Comment (bobsilverberg) [inline src/components/FullscreenGrid/index.tsx]: I know this is old code, but could this be an `enum` instead?

File: src/components/FullscreenGrid/index.tsx [hunk 2]
```diff
@@ -18,54 +28,14 @@ export const Header = ({
...
+export const ContentShell = ContentShellComponent;
 
-const FullscreenGrid = ({
-  children,
-  className,
-}: {
-  children?: AnyReactNode;
-  className?: string;
-}) => {
+export const FullscreenGridBase = ({ children, className }: Props) => {
```
Comment (bobsilverberg) [inline src/components/FullscreenGrid/index.tsx]: Out of interest, why did this need to be converted into a class component?

File: src/components/FullscreenGrid/index.tsx [hunk 3]
```diff
@@ -18,54 +28,14 @@ export const Header = ({
...
-      <main className={makeClassName(styles.content, className)}>
-        {children}
-      </main>
-      <aside className={makeClassName(styles.altSidePanel, altSidePanelClass)}>
-        {altSidePanel}
-      </aside>
-    </React.Fragment>
-  );
-};
+export const ContentShell = ContentShellComponent;
```
Comment (bobsilverberg) [inline src/components/FullscreenGrid/index.tsx]: I find this a bit odd. Why not just make `ContentShell` a component in its own right, as we've done with basically every other component. I realize that we don't ever imagine using it outside of a `FullscreenGrid`, but it still feels weird to have a component which is only accessible from within another component.

What's the advantage to doing it this way?

File: src/components/FullscreenGrid/index.tsx (hunk 2 above)
Comment (willdurand) [reply src/components/FullscreenGrid/index.tsx]: It's not a class-based component but I added the `Base` suffix as this is now a connected component. This is needed to retrieve the `fullGrid` reducer data.

File: src/components/FullscreenGrid/index.tsx (hunk 3 above)
Comment (willdurand) [reply src/components/FullscreenGrid/index.tsx]: mmh, this was mostly for backward compatibility. Let's move this component to `src/components` then.

File: src/components/FullscreenGrid/index.tsx (hunk 1 above)
Comment (willdurand) [reply src/components/FullscreenGrid/index.tsx]: Yes!

**Pull request merged**